src/api/rest/decorators/with-reducers.js

```diff
@@ -0,0 +1,129 @@
+import * as Actions from 'api/rest/actions';
```

Comment:
I'll be available this week-end to pair and walk you through it.

Here's some bits. I'm assuming that "pure" api call is a call made without the `withReducers()` wrapper.

Basically, adding a pure api call is as simple as defining the method in the client. (in the accurate namespace).

Here's client.activity.getEventsReceived() braked down:
```js
  activity = {
    // First arguments are the ones truly required to craft the URL that will be requested.
    // 
    // The last argument should ALWAYS be params, in all methods. It's referred to as
    // "magicalArg" in withReducers(). It's an object with the following keys (for now):
    //  - loadMore: boolean. Can be passed by the user to request the next page of results
    //  - url: string. This will be automatically provided by `withReducers` when paginating, 
    //                       and will contain the next page url. If present, then fetch()
    //                       will call this url instead of `users/${userId}/received_events`
    getEventsReceived: async (userId, params) => {
      // notice that we call THIS.fech() and not the native fetch()
      return this.fetch(
         // The url that should be initially requested
         // this.fetch() will chose to ignore it and use params.url instead if provided
        `users/${userId}/received_events`, 
        {
          // the schema to be used against the returned json to normalize it
          schema: Schemas.EVENT_ARRAY,
        },
        // always pass this down to this.fetch()
        params
      ).then(struct => {
        // this.fetch() returns a structure with the following keys
        // - response: The fetch() Response object
        // - schema: The schema that was passed to it
        return {
          ...struct,
          // if we're calling a paginated endpoint, automatically compute and return next page url
          nextPageUrl: this.getNextPageUrl(struct.response),
        };
      });
    },
  };
```

> Now you may wonder why we're passing schema to this.fetch(), and then retrieve it back? 
> I'm wondering too 🤣. I'll clean that up.
> 
> The thing is, we need to retrieve the schema to normalize against along with the response :
> - If used as a pure call, the developer may find handy to have the expected response, described by the schema. (optional)
> - When used with `withReducers()`, the proxy uses this schema to normalize the JSON.
 (required))

Second, if we intended to use this newly added pure api method with `withReducers()`, we have two extra steps:
1. define an action set in actions/ (with a specific naming: ACTIVITY_GET_EVENTS_RECEIVED)
2. if the call is to be paginated, add a reducer in pagination.js as follow:
     `ACTIVITY_GET_EVENTS_RECEIVED: paginate(Actions.ACTIVITY_GET_EVENTS_RECEIVED)`,

That's it.